install-app.sh

```
@@ -5,16 +5,17 @@
 # runs scripts under root user. 
 
 # install system updates and isntall python3-pip package using apt. '-yq' flags are 
-# used to suppress any interactive prompts - we won't be able to confirm operation 
-# when running the script as VM extention.  
+# used to suppress any interactive prompts - we won't be able to confirm operation
+# when running the script as VM extention.
 apt-get update -yq
 apt-get install python3-pip -yq
 
-# Create a directory for the app and download the files. 
-mkdir /app 
+
```

Add `set -euo pipefail` (or at least `set -e`) on an early line so the script exits on any error. The VM extension runs these scripts as root but without strict mode you may end up with partial installs and silent failures. Example to add before package operations: `set -euo pipefail`.